rotkehlchen/accounting/importer/json.py

```diff
@@ -0,0 +1,96 @@
+import json
+import logging
+from pathlib import Path
+from typing import Tuple
+
+from marshmallow import EXCLUDE, ValidationError
+from pysqlcipher3 import dbapi2 as sqlcipher
+
+from rotkehlchen.api.v1.schemas import (
+    HistoryBaseEntrySchema,
+    IgnoredActionsModifySchema,
+    ModifiableSettingsSchema,
+)
+from rotkehlchen.db.dbhandler import DBHandler
+from rotkehlchen.db.history_events import DBHistoryEvents
+from rotkehlchen.errors.misc import InputError
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class JSONImporter:
+    def __init__(self, db: DBHandler) -> None:
+        self.db = db
+
+    def import_history_debug(self, filepath: Path) -> Tuple[bool, str]:
+        """Imports the user events, settings & ignored actions identifiers for debugging."""
+        try:
+            with open(filepath) as f:
+                debug_data = json.load(f)
+        except (PermissionError, json.JSONDecodeError) as e:
+            return False, str(e)
+
+        has_required_keys = (
+            'events' in debug_data or
+            'settings' in debug_data or
+            'ignored_events_ids' in debug_data
+        )
+        if has_required_keys is False:
+            return False, 'import history data contains invalid keys'
+        has_required_types = (
+            isinstance(debug_data['events'], list) or
+            isinstance(debug_data['settings'], dict) or
+            isinstance(debug_data['ignored_events_ids'], dict)
+        )
+        if has_required_types is False:
+            return False, 'import history data contains invalid data types'
+
+        # add history events.
+        log.debug('Trying to add history events')
+        history_db = DBHistoryEvents(self.db)
+        try:
+            for event in debug_data['events']:
+                event = HistoryBaseEntrySchema(identifier_required=False).load(event)
```

Comment:
This will most likely not be an issue. This piece of code is only used in `rest.py`

Reply:
Hmhmhm my inside perfectionist still doesn't really like it tbh as `schemas.py` is used only in the `api` module right now and in tests. I think should be okay to keep it here but I want to ask Lefteris since this situation really intrigued me :smile:

Reply:
So after the discussion with Left everything is fine here! :partying_face:

Reply:
Yayyy. Pushing now ;)